### PR TITLE
Products view model

### PR DESCRIPTION
**Pull request by AbdulrahmanAli19**
Products screen has been created with a shimmer effect,
Screenshot (111)
Screenshot (112)

### CODE REVIEW

File: app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        viewModel.getProductsById(args.brandId)
+        viewModel.productsLiveData.observe(viewLifecycleOwner) {
+            when (it) {
+                is ResultState.Loading -> {
+                    binding.shimmer.root.showShimmer(true)
+                    binding.shimmer.root.startShimmer()
```
Comment (AhmedAbdallah125) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt:47]: use apply scope function to make code more readable

File: app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                }
+                is ResultState.Success -> {
+                    binding.shimmer.root.hideShimmer()
+                    binding.shimmer.root.stopShimmer()
+                    binding.shimmer.root.visibility = View.GONE
```
Comment (AhmedAbdallah125) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt:52]: use apply

File: app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            onUnLike = onUnLike
+        )
+        adapter.setProducts(products)
+        binding.productsRecycler.layoutManager =
+            GridLayoutManager(
+                requireContext(),
+                2,
+                RecyclerView.VERTICAL,
+                false
+            )
```
Comment (AhmedAbdallah125) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt:94]: use apply to make code readable and concise

File: app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/viewmodel/ProductsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            fetchResult(res.await())
+        }
+    }
+
+    private fun fetchResult(networkResponse: NetworkResponse<Products>) {
+        _productsMutableLivaData.value = ResultState.Loading
+        when (networkResponse) {
+            is NetworkResponse.SuccessResponse -> {
+                networkResponse.data.let {
+                    if (it.products != null && it.products.isNotEmpty()) {
```
Comment (AhmedAbdallah125) [inline app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/viewmodel/ProductsViewModel.kt:37]: you can use .!isNullOrEmpty()

**Review by AhmedAbdallah125 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt (hunk 3 above)
Comment (AbdulrahmanAli19) [reply app/src/main/java/com/iti/android/team1/ebuy/ui/productsScreen/view/ProductsFragment.kt:94]: done

**Pull request merged**